subprojects/launcher/src/main/java/org/gradle/launcher/daemon/configuration/BuildProcess.java

```diff
@@ -43,9 +43,19 @@ public boolean configureForBuild(DaemonParameters requiredBuildParameters) {
 
         final JvmOptions jvmOptions = new JvmOptions(new IdentityFileResolver());
         jvmOptions.systemProperties(getJvmOptions().getImmutableSystemProperties());
+        if (requiredBuildParameters.hasUserConfiguredImmutableJvmArgs()) {
```

Comment:
I'm curious why we don't just use `getJvmOptions().getAllImmutableJvmArgs()` instead of constructing a new JvmOptions object here.